### PR TITLE
Deduplicate code for adding custom RPM repositories for CentOS Stream

### PR DESCRIPTION
**Pull request by goarsna**
A separate procedure is used for adding custom RPM repositories for CentOS Stream. As this one is very similar to the generic procedure for adding custom RPM repositories, I propose reusing that one.

* [x] I am familiar with the [contributing](https://github.com/theforeman/foreman-documentation/blob/master/CONTRIBUTING.md) guidelines.

Please cherry-pick my commits into:

* [ ] Foreman 3.10/Katello 4.12
* [ ] Foreman 3.9/Katello 4.11 (planned Satellite 6.15)
* [ ] Foreman 3.8/Katello 4.10
* [ ] Foreman 3.7/Katello 4.9 (Satellite 6.14)
* [ ] Foreman 3.6/Katello 4.8
* [ ] Foreman 3.5/Katello 4.7 (Satellite 6.13; orcharhino 6.6/6.7)
* [ ] Foreman 3.4/Katello 4.6 (EL8 only)
* [ ] Foreman 3.3/Katello 4.5 on EL7 & EL8 (Satellite 6.12 on EL8 only; orcharhino 6.4/6.5 on EL8 only)
* We do not accept PRs for Foreman older than 3.3.

### CODE REVIEW

File: guides/common/modules/proc_adding-custom-rpm-repositories-el.adoc [hunk 1]
```diff
@@ -17,8 +17,10 @@ For more information, see xref:Creating_a_Custom_Product_{context}[].
 ** *Upstream URL*: `{yum_repo_baseos_url}`
 * *{yum_repo_appstream_name}*
 ** *Upstream URL*: `{yum_repo_appstream_url}`
+ifeval::["{os_name_anchor}" != "CentOS_Stream"]
```
Comment (goarsna) [inline guides/common/modules/proc_adding-custom-rpm-repositories-el.adoc:20]: I'm unsure about this check. Is there any better way to solve this?

Comment (maximiliankolb) [reply guides/common/modules/proc_adding-custom-rpm-repositories-el.adoc:20]: I think this is fine as is. Alternatively, you could check if there's also an "extras" repo for CentOS Stream 9.

Comment (goarsna) [reply guides/common/modules/proc_adding-custom-rpm-repositories-el.adoc:20]: I don't think there is an "extras" repository for CentOS Stream 9. At least I couldn't find any listed on https://mirror.stream.centos.org/9-stream/ and https://centos.pkgs.org/.

Comment (maximiliankolb) [reply guides/common/modules/proc_adding-custom-rpm-repositories-el.adoc:20]: There is:

````
$ podman run --rm -it centos:stream9

# dnf makecache
CentOS Stream 9 - BaseOS                                                                                                                                                      1.8 MB/s | 8.0 MB     00:04    
CentOS Stream 9 - AppStream                                                                                                                                                   2.2 MB/s |  19 MB     00:08    
CentOS Stream 9 - Extras packages                                                                                                                                              80 kB/s |  16 kB     00:00    
Last metadata expiration check: 0:00:01 ago on Wed Mar 20 15:07:50 2024.
Metadata cache created.
````

Comment (goarsna) [reply guides/common/modules/proc_adding-custom-rpm-repositories-el.adoc:20]: Thanks to @maximiliankolb I found out that I was searching at the wrong place.
The "extras" repository for CentOS Stream 9 is located at https://mirror.stream.centos.org/SIGs/9-stream/.

**Review by maximiliankolb — APPROVED** [review]
Thanks Goars, LGTM

**Comment by maximiliankolb** [comment]
Merged to "master" and cherry-picked:
   a7df21b54e..2122f308b1  3.10 -> 3.10
   a0069692e0..898693c62e  3.9 -> 3.9

**Pull request merged**